### PR TITLE
Issue 694: ReadHandle doesn't have getLength() method

### PR DESCRIPTION
**Pull request by sijie**
Descriptions of the changes in this PR:

- add `getLength()` to ReadHandle interface
- add `getLastAddPushed` to WriteHandle interface
- add unit tests and fix issue in MockBookKeeperTestCase

### CODE REVIEW

**Comment by asfgit** [comment]
SUCCESS 
 
--none--

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/client/api/WriteHandle.java [hunk 1]
```diff
@@ -52,4 +52,12 @@
         return append(Unpooled.wrappedBuffer(data));
     }
 
+    /**
+     * Get the entry id of the last entry that has been enqueued for addition (but
+     * may not have possibly been persisted to the ledger).
+     *
+     * @return the entry id of the last entry pushed or -1 if no entry has been pushed.
+     */
+    long getLastAddPushed();
```
Comment (eolivelli) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/client/api/WriteHandle.java]: Good to have this only here and not in WriteAdvHandle, it makes sense only here

**Comment by asfgit** [comment]
FAILURE
 
--none--

**Comment by asfgit** [comment]
FAILURE
 
--none--

**Pull request closed without merging**